src/ui_fsmenu/launch_mpg.cc

```diff
@@ -380,8 +300,35 @@ void FullscreenMenuLaunchMPG::think() {
 	refresh();
 
 	// unfocus chat window when other UI element has focus
-	if (!chat_->has_focus()) {
-		chat_->unfocus_edit();
+	if (!chat_.has_focus()) {
+		chat_.unfocus_edit();
+	}
+}
+
+void FullscreenMenuLaunchMPG::map_changed() {
+	const GameSettings& settings = settings_->settings();
+	if (!g_fs->file_exists(settings.mapfilename)) {
+		map_details.show_warning(
+		   _("The selected file can not be found. If it is not automatically transferred to you, "
+		     "please write to the host about this problem."));
+	} else {
+		// Care about the newly selected file. This has to be done here and not
+		// after selection of a new map / saved game, as the clients user
+		// interface can only notice the change after the host broadcasted it.
+		if (settings.savegame) {
+			load_previous_playerdata();
+		} else {
+			load_map_info();
+			if (settings.scenario) {
+				set_scenario_values();
+			}
+		}
+		// Try to translate the map name.
+		// This will work on every official map as expected
+		// and 'fail silently' (not find a translation) for already translated campaign map names.
+		// It will also translate 'false-positively' on any user-made map which shares a name with
+		// the official maps, but this should not be a problem to worry about.
+		i18n::Textdomain td("maps");
```

Comment:
This line does nothing at all because `td` goes out of scope immediately

Reply:
removed